cloudevents/extensions/expirytime.md

```diff
@@ -0,0 +1,58 @@
+# Expiry Time Extension
+
+This extension provides a mechanism to hint to [consumers](../spec.md#consumer)
+or [intermediaries](../spec.md#intermediary) a time after which an
+[event](../spec.md#event) may be ignored.
+
+In distributed systems with message delivery guarantees, events may be delivered
+to a consumer some significant amount of time after an event has been sent via
+an intermediary. In this situation, it may be desirable to ignore events that
+are no longer relevant. The [`time` attribute](../spec.md#time) may be used
+to handle this on the consumer side but can be tricky if the logic should vary
+depending on the event type or producer.
+
+## Attributes
+
+### expirytime
+
+- Type: `Timestamp`
+- Description: Timestamp indicating an event is no longer useful after the
+  indicated time.
+- Constraints:
+  - REQUIRED
+  - If present, MUST adhere to the format specified in
+    [RFC 3339](https://tools.ietf.org/html/rfc3339)
+  - SHOULD be equal to or later than the `time` attribute, if present
```

Comment:
I think it might make sense to add another one:
```
- consumers SHOULD ignore this event after the specified time
```

Also, should this apply to consumer or to intermediaries too?